extensions/amp-experiment/1.0/test/test-attribute-mutation.js

```diff
@@ -0,0 +1,201 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AttributeMutationDefaultClass} from '../mutation/attribute-mutation-default-class';
+import {AttributeMutationDefaultStyle} from '../mutation/attribute-mutation-default-style';
+import {AttributeMutationDefaultUrl} from '../mutation/attribute-mutation-default-url';
+import {toggleExperiment} from '../../../../src/experiments';
+
+describes.realWin(
+  'amp-experiment attribute-mutation-*',
+  {
+    amp: {
+      extensions: ['amp-experiment:1.0'],
+    },
+  },
+  env => {
+    let win, doc;
+
+    beforeEach(() => {
+      win = env.win;
+      doc = win.document;
+
+      toggleExperiment(win, 'amp-experiment-1.0', true);
+
+      doc.body.innerHTML = '';
+    });
+
+    function getAttributeMutationParamsObject(attributeName, value) {
+      return {
+        mutationRecord: {
+          'type': 'characterData',
+          'target': '.my-test-element-with-this-class',
+          'value': value,
+          'attributeName': attributeName,
+        },
+        elements: [doc.createElement('div'), doc.createElement('div')],
+      };
+    }
+
+    function getAttributeMutationDefaultClass(value) {
+      const paramsObject = getAttributeMutationParamsObject('class', value);
+      return new AttributeMutationDefaultClass(
+        paramsObject.mutationRecord,
+        paramsObject.elements
+      );
+    }
+
+    function getAttributeMutationDefaultStyle(value) {
+      const paramsObject = getAttributeMutationParamsObject('style', value);
+      return new AttributeMutationDefaultStyle(
+        paramsObject.mutationRecord,
+        paramsObject.elements
+      );
+    }
+
+    function getAttributeMutationDefaultUrl(value) {
+      const paramsObject = getAttributeMutationParamsObject('src', value);
+      return new AttributeMutationDefaultUrl(
+        paramsObject.mutationRecord,
+        paramsObject.elements
+      );
+    }
+
+    describe('validate', () => {
```

Comment:
Can we add a test that tests attributes other than `class` `style` `src/href` are not supported.